src/devices/gmmts.ts

```diff
@@ -0,0 +1,848 @@
+import {Definition, Fz, Tz, KeyValue, Zh} from '../lib/types';
+/* eslint-disable camelcase */
+/* eslint-disable max-len */
+/* eslint-disable no-multi-spaces */
+/* eslint-disable object-curly-spacing */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import * as exposes from '../lib/exposes';
+import * as globalStore from '../lib/store';
+import * as reporting from '../lib/reporting';
+import fz from '../converters/fromZigbee';
+import {repInterval} from '../lib/constants';
+
+import * as ota from '../lib/ota';
+import {Buffer} from 'buffer';
+import {Device} from 'zigbee-herdsman/dist/controller/model';
+import {logger} from '../lib/logger';
+import {Zcl} from 'zigbee-herdsman';
+
+const ea = exposes.access;
+const e = exposes.presets;
+
+const METER_ID_CLUSTER = 'haMeterIdentification'; // 0x0B01
+const ELEC_MES_CLUSTER = 'haElectricalMeasurement'; // 0x0B04
+const METERING_CLUSTER = 'seMetering'; // 0x0702
+const TICMETER_CLUSTER = 'manuSpecificGmmts'; // 0xFF42
+
+const STRING = 'string';
+const NUMBER = 'numeric';
+const NUM_RW = 'read/write numeric';
+const ENUM = 'enum';
+const TIME = 'date';
+
+const DEFAULT_POLL_INTERVAL = 120;
+
+const C = {
+    ANY: 'AUTO',
+    BASE: 'BASE',
+    HCHP: 'HCHP',
+    EJP: 'EJP',
+    TEMPO: 'TEMPO',
+};
+
+const E = {
+    ANY: 'AUTO',
+    MONO: 'MONOPHASE',
+    TRI: 'TRIPHASE',
+};
+
+const T = {
+    ANY: 'AUTO',
+    HIST: 'HISTORIQUE',
+    STD: 'STANDARD',
+};
+
+const mode_tic_enum = ['HISTORIQUE', 'STANDARD', 'AUTO'];
+const mode_elec_enum = ['MONOPHASE', 'TRIPHASE', 'AUTO'];
+const mode_contract_enum = ['AUTO', 'BASE', 'HCHP', 'EJP', 'TEMPO', 'PRODUCTEUR'];
+
+// prettier-ignore
+const ticmeter_options = [
+    e.numeric(`refresh_rate`, ea.SET).withValueMin(60).withDescription(`Temps d'actualisation des valeurs statiques (celles qui possèdent des boutons refresh). Par défaut: ${DEFAULT_POLL_INTERVAL} s`).withValueMin(60).withValueMax(3600),
```

Comment:
There is a lot of French here, please use English instead.

Reply:
This product is intended only for the French market (French electric meter). Is it possible to add translations?

Reply:
Translations are not support for this, but please translate it to English (ZLinky_TIC is also in English)

Reply:
I added an option to change the language.